skill-tree/src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java

```diff
@@ -59,4 +72,43 @@ public EndorsementModel createEndorsement(EndorsementDRO endorsementDRO) {
             throw new NoEntityException("Skill with id:" + skillId + " not found");
         }
     }
+
+    @Override
+    public GenericResponse<Void> updateEndorsementStatus(String id, String status) {
+        UserModel user =
+                (UserModel) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if (!user.getRole().equals(UserRole.SUPERUSER)) {
+            throw new InsufficientAuthenticationException(
+                    "Unauthorized, Access is only available to super users");
+        }
+        if (!(Objects.equals(status, EndorsementStatus.APPROVED.name())
+                || Objects.equals(status, EndorsementStatus.REJECTED.name()))) {
```

Comment:
If are referring to using ‘status == Endorsement.APPROVED.name()’ then I did would like to tell you that. As status is a String(captain S), Hence it is a wrapper class for string which can have ‘null’ as a value which could be a possibility. This is the main reason for using Objects.equals() here.
Also if you try to use ‘==‘ the IDE would itself suggest to use Objects.equals() as it is a standard in latest Java for comparing strings.
Do let me know if understood it incorrectly.

Reply:
So, you can use `status.equals(Endorsement.APPROVED.name())` instead of `==`. It will only match the values, not address.

Reply:
@Atifsid -> `==` will handle the null case. We don't use it because it compares the strings using string reference. To understand about strings in java, you can read these (https://www.baeldung.com/java-compare-strings, https://www.geeksforgeeks.org/string-initialization-java-string-literal-vs-string-object/).

Reply:
switched to string.equals()